pkg/driver/azure-disk/azure_disk.go

```diff
@@ -0,0 +1,411 @@
+	// sync config map with additional trust bundle to the operator namespace,
+	// so the operator can get it as a ConfigMap volume.
+	srcConfigMap := resourcesynccontroller.ResourceLocation{
+		Namespace: openshiftDefaultCloudConfigNamespace,
+		Name:      configMapName,
+	}
+	dstConfigMap := resourcesynccontroller.ResourceLocation{
+		Namespace: c.ControlPlaneNamespace,
+		Name:      hypershiftCloudConfigName,
+	}
+	cloudConfigSyncController := resourcesynccontroller.NewResourceSyncController(
+		c.OperatorClient,
+		c.KubeInformers,
+		c.KubeClient.CoreV1(),
+		c.KubeClient.CoreV1(),
+		c.EventRecorder)
+
+	err := cloudConfigSyncController.SyncConfigMap(dstConfigMap, srcConfigMap)
+	if err != nil {
+		return nil, err
+	}
+	return cloudConfigSyncController, nil
+}
+
+func getStackHubStorageClassHook() csistorageclasscontroller.StorageClassHookFunc {
+	return func(_ *opv1.OperatorSpec, sc *storagev1.StorageClass) error {
+		sc.Parameters["skuname"] = "Premium_LRS"
+		return nil
+	}
+}
+
+func runningInAzureStackHub(infra *opCfgV1.Infrastructure) bool {
+	if infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Azure != nil &&
+		infra.Status.PlatformStatus.Azure.CloudName == opCfgV1.AzureStackCloud {
+		return true
+	}
+	return false
+}
+
+// getKMSKeyHook checks for AzureCSIDriverConfigSpec in the ClusterCSIDriver object.
+// If it contains DiskEncryptionSet, it sets the corresponding parameter in the SC.
+// This allows the admin to specify a customer managed key to be used by default.
+func getKMSKeyHook(c *clients.Clients) csistorageclasscontroller.StorageClassHookFunc {
+	hook := func(_ *opv1.OperatorSpec, class *storagev1.StorageClass) error {
+		ccdLister := c.OperatorInformers.Operator().V1().ClusterCSIDrivers().Lister()
+		ccd, err := ccdLister.Get(class.Provisioner)
+		if err != nil {
+			return err
+		}
+
+		driverConfig := ccd.Spec.DriverConfig
+		if driverConfig.DriverType != opv1.AzureDriverType || driverConfig.Azure == nil {
+			klog.V(4).Infof("No AzureCSIDriverConfigSpec defined for %s", class.Provisioner)
+			return nil
+		}
+
+		encset := driverConfig.Azure.DiskEncryptionSet
+		if encset == nil {
+			klog.V(4).Infof("Not setting empty %s parameter in StorageClass %s", diskEncryptionSetID, class.Name)
+			return nil
+		}
+
+		if class.Parameters == nil {
+			class.Parameters = map[string]string{}
+		}
+		value := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Compute/diskEncryptionSets/%s", encset.SubscriptionID, encset.ResourceGroup, encset.Name)
+		klog.V(4).Infof("Setting %s = %s in StorageClass %s", diskEncryptionSetID, value, class.Name)
+		class.Parameters[diskEncryptionSetID] = value
+		return nil
+	}
+
+	// Explicitly instantiate ClusterCSIDriver informer, so it is synced during WaitForCacheSync
+	// and thus its lister is populated
+	// _ = c.OperatorInformers.Operator().V1().ClusterCSIDrivers().Informer()
+	return hook
```

Comment:
Do you think this is a bug in the current code? Should we have a bug to track it, since it probably affects other operators too?

Reply:
It is not a bug as such. @jsafrane used this trick to get informers to populated but I don't actually need this workaround because I explicitly added objects to informers before running tests and it worked just fine.  I can do the same aws-ebs operator and we won't need this anymore.